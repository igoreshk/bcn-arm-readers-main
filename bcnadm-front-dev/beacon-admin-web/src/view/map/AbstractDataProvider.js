/* eslint-disable no-unused-vars */
/* @flow*/

class AbstractDataProvider {
  /**
   * @param {string} buildingId
   * @param {string} levelId
   * @param {object} dto
   * @param {object} entityId
   * @returns {Promise}
   */
  getDtos(buildingId, levelId) {
    throw Error('Method not supported');
  }

  getDto(buildingId, levelId, markerId) {
    throw Error('Method not supported');
  }

  saveDto(buildingId, levelId, dto) {
    throw Error('Method not supported');
  }

  removeDto(buildingId, levelId, dto) {
    throw Error('Method not supported');
  }

  removeAll(buildingId, levelId) {
    throw Error('Method not supported');
  }
}

export default AbstractDataProvider;
