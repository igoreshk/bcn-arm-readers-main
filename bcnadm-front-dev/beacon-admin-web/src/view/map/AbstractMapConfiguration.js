/* eslint-disable no-unused-vars */
/* @flow*/

export default class AbstractMapConfiguration {
  /**
   * createMap
   * @returns {Promise}
   */
  createMap() {
    throw Error('Method not supported');
  }

  /**
   * setMapBackground
   * @param {object} map
   * @returns {void}
   */
  setMapBackground(map) {
    throw Error('Method not supported');
  }

  /**
   * getImageAndCalculateBounds
   * @param {object} map
   * @param {string} levelId
   * @param {string} buildingId
   * @returns {Promise}
   */
  getImageAndCalculateBounds(map, levelId, buildingId) {
    throw Error('Method not supported');
  }

  /**
   * configureOverlay
   * @param {object} map
   * @param {object} bounds
   * @param {string} image
   * @returns {Promise}
   */
  configureOverlay(map, bounds, image) {
    throw Error('Method not supported');
  }

  /**
   * setMapDraggingListener
   * @param {object} map
   * @param {object} bounds
   */
  setMapDraggingListener(map, bounds) {
    throw Error('Method not supported');
  }
}
