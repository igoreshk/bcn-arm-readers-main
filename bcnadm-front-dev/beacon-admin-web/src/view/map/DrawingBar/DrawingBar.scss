.drawing-bar__main-body {
  display: flex;
  position: relative;
}

.drawing-bar__wrapper {
  display: flex;
  position: absolute;
  left: 50%;
  top: 70px;
  transform: translate(-50%, -50%);
  z-index: 100;
}

.drawing-bar-button-group__button:nth-of-type(1) {
  z-index: 100;
  position: absolute;
  left: 50%;
  font-size: 10px;
  height: 30px;
  padding: 0;
  margin: 0;
  top: 100px;

  &:disabled {
    background-color: #77889a;
    cursor: not-allowed;
  }

  .MuiButton-label {
    margin: 0;
    padding: 0;
    width: 30px;
  }
}

.drawing-bar__name-input-bar {
  display: flex;
  width: 300px;
  position: absolute;
  top: 120px;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 2;
  justify-content: flex-end;
  align-items: center;
}

.drawing-bar__name-input-bar_input {
  display: flex;
  font-size: 7px;
  background-color: #FFF;
  width: 165px;

  .MuiInputBase-input {
    font-size: 16px;

    &::placeholder {
      padding-left: 20px;
      font-size: 14px;
    }
  }

  .MuiInputBase-root {
    padding: 5px 0 5px 10px;
  }
}

.drawing-bar__name-input-bar_button {
  width: 100px;
  height: 40px;
  
  .MuiButton-label {
    font-size: 8px;
  }
}

.drawing-bar__polygon-button,
.drawing-bar__polygon-send-button {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 80px;
  height: 30px;
  padding: 10px;
  align-self: center;
  cursor: pointer;
  background-color: #ff4d2c;
  border: none;
  border-radius: 7px;
  margin-top: 5px;

  &:hover {
    background-color: #3fe71d;
    font-weight: 600;
  }
}

.drawing-bar__polygon-send-button {
  margin-top: 10px;
  background-color: #0a81ec;

  &:hover {
    background-color: #f8d806;
    font-weight: 600;
  }
}

.leaflet-popup-content {
  font-size: 15px;
  text-transform: capitalize;
}

.leaflet-popup-content-wrapper {
  min-width: 120px;
}

.leaflet-container a.leaflet-popup-close-button {
  color: black;
}

.drawing-bar-button-group__error-bar {
  display: flex;
  z-index: 100;
  position: absolute;
  font-size: 10px;
  height: 30px;
  padding: 10px;
  margin: 0;
  top: 210px;
  border-radius: 7px;
  justify-content: space-between;
  align-items: center;
  gap: 5px;
}

.drawing-bar-button-group__error-bar_button {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  width: 30px;
  height: 20px;
  border: 1px solid #000000;
}

.drawing-bar__polygon-pop-up {
  display: flex;
  background-color: #FFFFFF;
  font-size: 25px;
  justify-content: center;
  align-items: center;
}

.drawing-bar__polygon-button_get-data {
  display: flex;
  font-size: 20px;
  cursor: pointer;
  background-color: #0a81ec;
  border: none;
  padding: 10px;
  border-radius: 7px;
}

.drawing-bar__polygon-pop-up-name {
  display: flex;
  font-size: 20px;
  justify-content: center;
  text-transform: capitalize;
  margin-bottom: 5px;
}