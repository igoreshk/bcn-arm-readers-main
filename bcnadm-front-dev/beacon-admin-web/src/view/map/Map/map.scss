.mapWrapper {
  flex: 1 0;
  height: calc(100vh - 56px);
  z-index: 1;
}
.mapContainer {
  height: 100%;
}
