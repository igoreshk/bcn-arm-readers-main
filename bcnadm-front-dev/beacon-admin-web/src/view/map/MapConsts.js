/**
 * Contains set of constants for configuring map
 * @namespace mapConsts
 * @prop {object} mapOptions Constant, used in google.maps.Map() constructor when creating map
 * @prop {function} configureCoordMapType Configures overlay map type & map tiles
 * (See: {@link configureCoordMapType})
 * @prop {function} configureOverlay Configures overlay, particularly sets custom image as an overlay, size of an overlay etc
 * (See: {@link configureOverlay})
 *
 * @see [Custom overlays]{@link https://developers.google.com/maps/documentation/javascript/customoverlays?hl=en}
 * To read about configuring custom overlays
 */
export const INITIAL_ZOOM = 18;

const mapConsts = {
  mapOptions: {
    zoom: INITIAL_ZOOM,
    minZoom: 12,
    maxZoom: 28,
    center: { lat: 0, lng: 0 },
    mapTypeId: 'roadmap',
    zoomControl: true,
    mapTypeControl: false,
    scaleControl: true,
    streetViewControl: false,
    rotateControl: false,
    fullscreenControl: false
  },
  /**
   * This function configures tiles of map
   * @function
   * @param coord {object} Coordinates on current tile
   * @param zoom {number} Zoom rate
   * @param ownerDocument {object} returns higher-level document object for current node
   * @returns {object} configured HTML element
   */
  divConfig(coord, zoom, ownerDocument) {
    const div = ownerDocument.createElement('div');
    div.innerHTML = coord;
    div.style.width = `${this.tileSize.width}px`;
    div.style.height = `${this.tileSize.height}px`;
    div.style.fontSize = '0';
    div.style.borderStyle = 'solid';
    div.style.borderWidth = '10000px';
    div.style.borderColor = '#e5e5e5';
    return div;
  },
  /**
   * This function is called when the map is ready for the overlay to be attached
   *@function
   */
  onAdd() {
    const div = document.createElement('div');
    div.style.border = 'none';
    div.style.borderWidth = '0px';
    div.style.position = 'absolute';

    const img = document.createElement('img');
    img.src = this.image;
    img.style.width = '100%';
    img.style.height = '100%';
    div.appendChild(img);

    this.div = div;
    const panes = this.getPanes();
    panes.overlayLayer.appendChild(div);
  },
  /**
   * This function  will be called when the object (custom overlay) is first displayed
   * @function
   */
  draw() {
    const overlayProjection = this.getProjection();
    const sw = overlayProjection.fromLatLngToDivPixel(this.bounds.getSouthWest());
    const ne = overlayProjection.fromLatLngToDivPixel(this.bounds.getNorthEast());
    const div = this.div;
    this.div.id = 'layer';
    div.style.left = `${sw.x}px`;
    div.style.top = `${ne.y}px`;
    div.style.width = `${ne.x - sw.x}px`;
    div.style.height = `${sw.y - ne.y}px`;
    div.style.transform = `rotate(${this.rotation}rad)`;
  },
  /**
   * This function is used to clean up any elements added within the overlay.
   * @function
   */
  onRemove() {
    this.div.parentNode.removeChild(this.div);
    this.div = null;
  }
};
export const TILE_SIZE = 1000;
export default mapConsts;
/**
 * Mode when user adds beacon markers
 * @constant
 * @type {string}
 */
const BEACON_MODE = 'beacons';
/**
 * Mode when user adds reader markers
 * @constant
 * @type {string}
 */
const READER_MODE = 'readers';
/**
 * Mode when user adds vertex markers
 * @constant
 * @type {string}
 */
const ROUTE_MODE = 'routing';
/**
 * Mode when user adds area markers
 * @constant
 * @type {string}
 */
const AREAS_MODE = 'areas';
/**
 * Mode when user monitore people on the map
 * @constant
 * @type {string}
 */
const MONITORING_MODE = 'monitoring';
/**
 * Mode when user is able to draw on the map tile
 * @constant
 * @type {string}
 */
const DRAWING_MODE = 'drawing';
/**
 * Represents state of a dialog, where beacon's info can be configures
 * @constant
 * @type {string}
 */
const BEACON_DIALOG_OPEN = 'BEACON_DIALOG_OPEN';
/**
 * Represents state of a dialog, where beacon's info can be configures
 * @constant
 * @type {string}
 */
const BEACON_DIALOG_CLOSE = 'BEACON_DIALOG_CLOSE';
/**
 * Represents state of a dialog, where area's info can be configures
 * @constant
 * @type {string}
 */
const AREAS_DIALOG_OPEN = 'AREAS_DIALOG_OPEN';

const READER_DIALOG_OPEN = 'READER_DIALOG_OPEN';

const READER_DIALOG_CLOSE = 'READER_DIALOG_CLOSE';

const DISTANCE_DIALOG_OPEN = 'DISTANCE_DIALOG_OPEN';

const DISTANCE_DIALOG_CLOSE = 'DISTANCE_DIALOG_CLOSE';

/**
 * Represents state of a dialog, where area's info can be configures
 * @constant
 * @type {string}
 */
const AREAS_DIALOG_CLOSE = 'AREAS_DIALOG_CLOSE';

/**
 * Describes different modes
 * @type {{BEACON_MODE: string, ROUTE_MODE: string, AREAS_MODE: string}}
 */
const SCALE_MODE = 'scaling';

const GOOGLE = 'Google';

const OSM = 'OSM';

export const modes = {
  READER_MODE,
  BEACON_MODE,
  ROUTE_MODE,
  AREAS_MODE,
  SCALE_MODE,
  MONITORING_MODE,
  DRAWING_MODE
};
/**
 * Set of area/beacon dialog states
 * @constant
 * @type {object}
 */
export const dialog = {
  BEACON_DIALOG_OPEN,
  BEACON_DIALOG_CLOSE,
  AREAS_DIALOG_OPEN,
  AREAS_DIALOG_CLOSE,
  DISTANCE_DIALOG_OPEN,
  DISTANCE_DIALOG_CLOSE,
  READER_DIALOG_OPEN,
  READER_DIALOG_CLOSE
};

export const mapProviders = {
  GOOGLE,
  OSM
};
