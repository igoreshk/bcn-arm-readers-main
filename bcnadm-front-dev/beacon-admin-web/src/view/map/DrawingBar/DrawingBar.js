import React, { useEffect, useRef, useState } from 'react';
import L from 'leaflet';
import { Button, IconButton, Snackbar, SnackbarContent, TextField } from '@material-ui/core';
import PropTypes from 'prop-types';
import './DrawingBar.scss';
import { useHistory } from 'react-router-dom';
import { withLocalize } from 'react-localize-redux';
import { ModeButtonGroup } from 'view/map/ModeButtonGroup';
import { OnMapsDrawingService } from 'service/OnMapsDrawingService';
import { modes } from 'view/map/MapConsts';
import CloseIcon from '@material-ui/icons/Close';

export const googleLayerForOSMap = (
  pictureCoords,
  loadedOSMap,
  buildingImage,
  zoomPosition,
  buildingCenterCoordinates
) => {
  const bounds = L.latLngBounds([pictureCoords[0]]);
  const zoomValue = 18;
  loadedOSMap.current = L.map('map', { zoomControl: false }).setView(buildingCenterCoordinates, zoomValue);
  L.tileLayer('https://{s}.google.com/vt/lyrs=m&x={x}&y={y}&z={z}', {
    id: 'main',
    maxZoom: 18,
    subdomains: ['mt0', 'mt1', 'mt2', 'mt3'],
    zIndex: 3000
  }).addTo(loadedOSMap.current);
  zoomPosition.addTo(loadedOSMap.current);
  L.imageOverlay(buildingImage, pictureCoords).addTo(loadedOSMap.current);
  loadedOSMap.current?.fitBounds(bounds);
};

export const shapesRender = (figures, match, loadedOSMap, translate) => {
  figures.data?.map((figure) => {
    if (figure.levelId === match.params.level) {
      const lay = L.featureGroup().addTo(loadedOSMap?.current);
      const removePolygon = (event) => {
        const element = document.querySelectorAll('.drawing-bar__polygon-button');
        element.forEach((el) =>
          el.addEventListener('click', () => {
            OnMapsDrawingService.deleteShapeFromDataBase(figure.name);
            lay.removeLayer(event.sourceTarget._leaflet_id);
          })
        );
      };
      const showPopUpMessage = () => {
        const element = document.querySelectorAll('.drawing-bar__polygon-button');
        return element.forEach((x) => (x.textContent = translate('drawing.remove')));
      };
      const extractedCoordinates = Object.values(figure.coordinates).map((x) => [x.latitude, x.longitude]);
      const polygon = L.polygon(extractedCoordinates).addTo(lay);
      if (match.params.layer === modes.DRAWING_MODE) {
        polygon.bindPopup(`${'<div class="drawing-bar__polygon-pop-up">'}${figure.name}
          <button type="button" class="drawing-bar__polygon-button"></button></div>`);
      } else {
        polygon.bindPopup(
          match.params.layer === 'monitoring'
            ? `${'<div class="drawing-bar__polygon-pop-up"></div>'}<span class="drawing-bar__polygon-pop-up-name">${
                figure.name
              }</span>
          <button type="button" id="shape-getter" class="drawing-bar__polygon-button_get-data" value='${
            figure.entityId
          }'>${translate('monitoring.getVisitors')}</button>`
            : `${'<div class="drawing-bar__polygon-pop-up"></div>'}<span class="drawing-bar__polygon-pop-up-name">${
                figure.name
              }</span>`
        );
      }
      polygon.on('click', (event) => {
        removePolygon(event);
        showPopUpMessage();
      });
      return polygon;
    }
    return figure;
  });
};

export const zoomPosition = L.control.zoom({
  position: 'bottomright'
});

export const DrawingBar = ({ loadedOSMap, buildingCenterCoordinates, pictureCoords, picture, translate, match }) => {
  const layerWithMap = useRef();
  const [polygons, setPolygons] = useState([]);
  const [drawing, setDrawing] = useState(false);
  const ZOOM_AND_SCALE_VALUES = { zoom: 18, scale: 10 };
  const history = useHistory();
  const [coordinatesToSend, setCoordinatesToSend] = useState([]);
  const [sendingStatus, setSendingStatus] = useState({ error: false, sent: false });

  const sendPolygonsToDataBase = (data) => {
    const element = document.querySelectorAll('.drawing-bar__polygon-send-button');
    element.forEach((el) =>
      el.addEventListener('click', () => {
        OnMapsDrawingService.sendShapesToDataBase(data);
      })
    );
  };

  useEffect(() => {
    const closeButton = document.querySelector('.leaflet-popup-close-button');
    if (sendingStatus.sent === true) {
      closeButton.click();
    }
  }, [sendingStatus.sent]);

  useEffect(() => {
    if (loadedOSMap?.current) {
      layerWithMap.current = loadedOSMap.current;
    }
  }, [loadedOSMap, buildingCenterCoordinates, pictureCoords, picture]);
  const takenCoordinates = [];
  const [name, setName] = useState({ name: '', state: false, savedName: '' });
  const [dataForLocalStorage, setDataForLocalStorage] = useState([]);

  let map = layerWithMap.current;

  const onMapClick = () => {
    if (!map) {
      map = L.map('map', { zoomControl: false }).setView(buildingCenterCoordinates, ZOOM_AND_SCALE_VALUES.zoom);
    }
    if (!zoomPosition) {
      zoomPosition.addTo(map);
    }

    const layers = [];
    const coords = [];

    const activePolyline = L.featureGroup().addTo(map);
    const passivePolyline = L.featureGroup().addTo(map);
    map?.on('click', (evt) => {
      const coordinate = { latitude: evt.latlng.lat, longitude: evt.latlng.lng };
      takenCoordinates.push(coordinate);
      const mainLayer = L.featureGroup().addTo(map);
      const marker = L.marker([evt.latlng.lat, evt.latlng.lng]).addTo(activePolyline).addTo(passivePolyline);
      coords.push([evt.latlng.lat, evt.latlng.lng]);
      layers.push(marker);

      const polyline = L.polyline(coords).addTo(activePolyline);
      marker.on('click', () => {
        const deletePolygon = (event) => {
          const element = document.querySelectorAll('.drawing-bar__polygon-button');
          element.forEach((el) =>
            el.addEventListener('click', () => {
              mainLayer.removeLayer(event.sourceTarget._leaflet_id);
            })
          );
        };

        const popUpContent = `${'<div class="drawing-bar__polygon-name"></div>'}${name.savedName}
          <button type="button" class="drawing-bar__polygon-button">${translate('drawing.remove')}</button>
          <button type="button" class="drawing-bar__polygon-send-button">${translate('drawing.send')}</button>
        `;
        const polygon = L.polygon(coords).addTo(mainLayer);
        if (polygon) {
          setPolygons([...polygons, polygon]);
          map?.off('click');
          passivePolyline.clearLayers(layers);
          activePolyline.clearLayers(layers);
          setCoordinatesToSend([...coordinatesToSend, { takenCoordinates, name: name.savedName }]);
          setDataForLocalStorage([
            ...dataForLocalStorage,
            { coordinates: takenCoordinates, name: name.savedName, levelId: match.params.level }
          ]);
          setDrawing(false);
          setSendingStatus(sendingStatus);
        }
        polygon.bindPopup(popUpContent).openPopup();

        const dataToSend = {
          coordinates: takenCoordinates,
          name: name.savedName,
          levelId: match.params.level,
          status: sendingStatus,
          errorHandler: setSendingStatus
        };

        document.querySelectorAll('.drawing-bar__polygon-button').forEach((button) =>
          button.addEventListener('click', () => {
            mainLayer.removeLayer(polygon._leaflet_id);
          })
        );
        document.querySelectorAll('.drawing-bar__polygon-send-button').forEach((button) =>
          button.addEventListener('click', () => {
            OnMapsDrawingService.sendShapesToDataBase(dataToSend);
          })
        );
        polygon.on('click', (event) => {
          deletePolygon(event);
          sendPolygonsToDataBase(dataToSend);
        });
      });
      setName({ ...name, savedName: '', state: false });
      marker.on('contextmenu', () => {
        if (marker === layers.at(-1)) {
          layers.pop();
          activePolyline.removeLayer(marker);
          coords.pop();
          map?.removeLayer(polyline);
        }
      });
    });
  };

  const onNameSubmit = () => {
    setName({
      ...name,
      savedName: name.name,
      state: true,
      name: ''
    });
    setDrawing(true);
  };

  useEffect(() => {
    if (drawing === true) {
      onMapClick();
    }
  }, [drawing]);

  return (
    <div className="drawing-bar__main-body">
      <div className="drawing-bar__wrapper">
        <ModeButtonGroup
          match={match}
          translate={translate}
          history={history}
          scaleDistance={ZOOM_AND_SCALE_VALUES.scale}
        />
      </div>
      {sendingStatus.error === true && (
        <Snackbar
          open
          autoHideDuration={5000}
          className="drawing-bar-button-group__error-bar"
          onClose={() => setSendingStatus({ ...sendingStatus, error: false })}
        >
          <SnackbarContent
            className="drawing-bar-button-group__error-bar-content"
            message={translate('drawing.existingNameError')}
            action={
              <IconButton
                size="small"
                aria-label="close"
                color="inherit"
                onClick={() => setSendingStatus({ ...sendingStatus, error: false })}
              >
                <CloseIcon fontSize="small" />
              </IconButton>
            }
          />
        </Snackbar>
      )}
      {sendingStatus.sent === true && (
        <Snackbar
          open
          autoHideDuration={5000}
          className="drawing-bar-button-group__error-bar"
          onClose={() => setSendingStatus({ ...sendingStatus, sent: false })}
        >
          <SnackbarContent
            className="drawing-bar-button-group__error-bar-content"
            message={translate('drawing.successfulSending')}
            action={
              <IconButton
                size="small"
                aria-label="close"
                color="inherit"
                onClick={() => setSendingStatus({ ...sendingStatus, sent: false })}
              >
                <CloseIcon fontSize="small" />
              </IconButton>
            }
          />
        </Snackbar>
      )}
      {!name.savedName.length && name.state === false && (
        <div className="drawing-bar__name-input-bar">
          <TextField
            disabled={drawing}
            placeholder={translate('drawing.nameTyping')}
            className="drawing-bar__name-input-bar_input"
            value={name.name}
            type="text"
            onChange={(event) => setName({ ...name, name: event.target.value })}
            onKeyPress={(event) => event.key === 'Enter' && onNameSubmit()}
          />
          <Button
            className="drawing-bar__name-input-bar_button"
            color="primary"
            variant="contained"
            type="button"
            disabled={name.name.length === 0}
            onClick={() => {
              onNameSubmit();
            }}
          >
            {translate('drawing.nameConfirmation')}
          </Button>
        </div>
      )}
    </div>
  );
};

DrawingBar.propTypes = {
  picture: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Array)]),
  loadedOSMap: PropTypes.oneOfType([PropTypes.object.isRequired, PropTypes.instanceOf(Array).isRequired]),
  buildingCenterCoordinates: PropTypes.instanceOf(Array),
  pictureCoords: PropTypes.instanceOf(Array),
  translate: PropTypes.func,
  match: PropTypes.object
};

export default withLocalize(DrawingBar);
