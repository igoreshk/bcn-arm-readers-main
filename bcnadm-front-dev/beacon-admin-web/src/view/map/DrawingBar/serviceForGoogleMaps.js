import { googleLayerForOSMap, shapesRender, zoomPosition } from 'view/map/DrawingBar/DrawingBar';
import { mapProviders, modes } from 'view/map/MapConsts';

export const setMapForGoogle = (props, buildingImage) => {
  const {
    selectedMapProviderName,
    loadedOSMap,
    match,
    buildingCenterCoordinates,
    pictureCoords,
    translate,
    google,
    selectedMap,
    shapes
  } = props;

  if (selectedMapProviderName === mapProviders.GOOGLE && match.params.layer !== modes.DRAWING_MODE) {
    shapes.data?.map((figure) => {
      const polygon = new google.maps.Polygon({
        paths: figure.coordinates
          .map((coordinate) => [{ lat: coordinate.latitude, lng: coordinate.longitude }])
          .flat(1),
        strokeColor: '#3388ff',
        strokeOpacity: 0.8,
        strokeWeight: 2,
        fillColor: '#3388ff',
        fillOpacity: 0.05,
        editable: false,
        draggable: false
      });

      const infoWindow = new google.maps.InfoWindow();

      google.maps.event.addListener(polygon, 'click', (event) => {
        infoWindow.setContent(
          match.params.layer === modes.MONITORING_MODE
            ? `${'<div class="drawing-bar__polygon-pop-up"></div>'}<span class="drawing-bar__polygon-pop-up-name">${
                figure.name
              }</span>
          <button type="button" id="shape-getter" class="drawing-bar__polygon-button_get-data" value='${
            figure.entityId
          }'>${translate('monitoring.getVisitors')}</button>`
            : `${'<div class="drawing-bar__polygon-pop-up"></div>'}<span class="drawing-bar__polygon-pop-up-name">${
                figure.name
              }</span>`
        );
        infoWindow.setPosition(event.latLng);
        infoWindow.open(polygon);
      });

      if (figure.levelId === match.params.level) {
        polygon.setMap(selectedMap);
      }
      return polygon;
    });
  }

  if (
    selectedMapProviderName === mapProviders.GOOGLE &&
    !loadedOSMap.current &&
    match.params.layer === modes.DRAWING_MODE &&
    buildingCenterCoordinates &&
    pictureCoords.length !== 0
  ) {
    googleLayerForOSMap(pictureCoords, loadedOSMap, buildingImage, zoomPosition, buildingCenterCoordinates);
    shapesRender(shapes, match, loadedOSMap, translate);
  }
};
