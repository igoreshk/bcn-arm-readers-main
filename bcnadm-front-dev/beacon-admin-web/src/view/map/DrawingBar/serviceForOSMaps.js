import { withLocalize } from 'react-localize-redux';
import { shapesRender } from 'view/map/DrawingBar/DrawingBar';

export const setMapForOSM = (props) => {
  const { loadedOSMap, match, shapes, translate } = props;
  if (loadedOSMap.current) {
    try {
      shapesRender(shapes, match, loadedOSMap, translate);
    } catch (error) {
      console.error(error);
    }
  }
};

export default withLocalize(setMapForOSM);
