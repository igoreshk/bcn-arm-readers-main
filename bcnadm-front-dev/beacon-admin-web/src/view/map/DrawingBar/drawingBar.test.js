import React from 'react';
import { shallow } from 'enzyme';
import { DrawingBar } from 'view/map/DrawingBar/DrawingBar';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { modes } from 'view/map/MapConsts';

const match = {
  path: '/buildings/:building/levels/:level/:layer/:edit?',
  url: '/buildings/6127807a5c161c2136ae8cb4/levels/614085264bb5bb44010a68cb/drawing',
  isExact: true,
  params: {
    building: '6127807a5c161c2136ae8cb4',
    level: '614085264bb5bb44010a68cb',
    layer: modes.DRAWING_MODE
  }
};

const buildingCenterCoordinates = [57.2, 19.1];

const pictureCoords = [
  [{ lat: 57.97220388411178, lng: 19.271508080117208 }],
  [{ lat: 57.97535120449788, lng: 19.263594881636408 }]
];

const translate = jest.fn();

const picture = 'http://localhost:3000/api/v1/levels/614085264bb5bb44010a68cb/image';

const loadedOSMap = {
  current: {
    provider: 'OSM',
    options: {
      'center': [57.97377757885401, 19.267551480876808],
      'zoom': 18
    },
    _lastCenter: {
      'lat': 57.973066416050386,
      'lng': 19.265937209129337
    }
  }
};

const propsOSM = {
  loadedOSMap,
  selectedMapProviderName: 'OSM',
  match,
  translate
};
const propsGoogle = {
  buildingCenterCoordinates,
  pictureCoords,
  picture,
  selectedMapProviderName: 'Google',
  match,
  translate
};

const store = configureStore();

describe('renders DrawingBar with Google map', () => {
  const wrapperForGoogle = shallow(
    <Provider store={store({})}>
      <BrowserRouter>
        <DrawingBar {...propsGoogle} />
      </BrowserRouter>
    </Provider>
  );
  const containerForGoogle = wrapperForGoogle.find(DrawingBar);

  it('Checks that props with Google map are passed properly', () => {
    expect(containerForGoogle.props()).toEqual(propsGoogle);
  });
});

describe('Check matching snapshot with Google map', () => {
  const wrapperForGoogle = shallow(
    <Provider store={store({})}>
      <DrawingBar {...propsGoogle} />
    </Provider>
  );
  it('Check matching snapshot', () => expect(wrapperForGoogle).toMatchSnapshot());
});

describe('renders DrawingBar with OSM map', () => {
  const wrapperForOsm = shallow(
    <Provider store={store({})}>
      <BrowserRouter>
        <DrawingBar {...propsOSM} />
      </BrowserRouter>
    </Provider>
  );
  const containerForOsm = wrapperForOsm.find(DrawingBar);

  it('Checks that props with OSMap are passed properly', () => {
    expect(containerForOsm.props()).toEqual(propsOSM);
  });
});

describe('Check matching snapshot with OSM map', () => {
  const wrapperForOsm = shallow(
    <Provider store={store({})}>
      <DrawingBar {...propsOSM} />
    </Provider>
  );
  it('Check matching snapshot', () => expect(wrapperForOsm).toMatchSnapshot());
});
