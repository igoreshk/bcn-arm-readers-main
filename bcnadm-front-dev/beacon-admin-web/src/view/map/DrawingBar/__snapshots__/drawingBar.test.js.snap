// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check matching snapshot with Google map Check matching snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <DrawingBar
    buildingCenterCoordinates={
      Array [
        57.2,
        19.1,
      ]
    }
    match={
      Object {
        "isExact": true,
        "params": Object {
          "building": "6127807a5c161c2136ae8cb4",
          "layer": "drawing",
          "level": "614085264bb5bb44010a68cb",
        },
        "path": "/buildings/:building/levels/:level/:layer/:edit?",
        "url": "/buildings/6127807a5c161c2136ae8cb4/levels/614085264bb5bb44010a68cb/drawing",
      }
    }
    picture="http://localhost:3000/api/v1/levels/614085264bb5bb44010a68cb/image"
    pictureCoords={
      Array [
        Array [
          Object {
            "lat": 57.97220388411178,
            "lng": 19.271508080117208,
          },
        ],
        Array [
          Object {
            "lat": 57.97535120449788,
            "lng": 19.263594881636408,
          },
        ],
      ]
    }
    selectedMapProviderName="Google"
    translate={[MockFunction]}
  />
</ContextProvider>
`;

exports[`Check matching snapshot with OSM map Check matching snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <DrawingBar
    loadedOSMap={
      Object {
        "current": Object {
          "_lastCenter": Object {
            "lat": 57.973066416050386,
            "lng": 19.265937209129337,
          },
          "options": Object {
            "center": Array [
              57.97377757885401,
              19.267551480876808,
            ],
            "zoom": 18,
          },
          "provider": "OSM",
        },
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {
          "building": "6127807a5c161c2136ae8cb4",
          "layer": "drawing",
          "level": "614085264bb5bb44010a68cb",
        },
        "path": "/buildings/:building/levels/:level/:layer/:edit?",
        "url": "/buildings/6127807a5c161c2136ae8cb4/levels/614085264bb5bb44010a68cb/drawing",
      }
    }
    selectedMapProviderName="OSM"
    translate={[MockFunction]}
  />
</ContextProvider>
`;
